source/virtualBuffers/__init__.py

```diff
@@ -109,14 +109,30 @@ def obj(self):
 
 	@property
 	def label(self):
-		if self.itemType == "landmark":
-			attrs = self.textInfo._getControlFieldAttribs(self.vbufFieldIdentifier[0], self.vbufFieldIdentifier[1])
+		attrs = self.textInfo._getControlFieldAttribs(self.vbufFieldIdentifier[0], self.vbufFieldIdentifier[1])
+		roleRaw = attrs["role"]
+		role = controlTypes.roleLabels[roleRaw]
+		value = super(VirtualBufferQuickNavItem,self).label
+		# Translators: Reported label in the elements list for an element which which has no name and value
+		unlabeled = _("Unlabeled")
+		if self.itemType is not "heading":
 			name = attrs.get("name", "")
-			if name:
-				name += " "
-			return name + aria.landmarkRoles[attrs["landmark"]]
+			realStates=attrs["states"]
+			positiveStates = " ".join(controlTypes.stateLabels[st] for st in controlTypes.processPositiveStates(roleRaw, realStates, controlTypes.REASON_ELEMENTSLIST, realStates))

```

Comment:
This states stuff seems a bit arbitrary at the moment.

* Do we really need states at all here? Sure, I guess this means you get to hear "check box not checked" or the like, but I'm not convinced we need that in the Elements List anyway. If the user is using the list, they are trying to find a specific control. If they want an overview of the form's values, they should be using quick navigation, no?
* I'm a bit bothered by introducing REASON_ELEMENTSLIST when it's really only used for states calculation. Furthermore, we just use it to kill the collapsed state... but why the collapsed state? If we really do want specific states, I feel we should perhaps be explicit about that. Otherwise, we could just be continually adding more and more exceptions in controlTypes.
* Perhaps we can answer both of these questions if you can give me some specific use cases you had in mind here.
* At the very least, we'd need some comments here explaining why these choices were made.